util.py

```diff
@@ -0,0 +1,12 @@
+def convert_str_to_bool(value: str, default: bool) -> bool:
+    try:
+        if value is None:
+            return default
+        if value.lower() == 'true':
+            return True
+        elif value.lower() == 'false':
+            return False
```

Comment:
Potential logical inconsistency with 'none' value
The value 'none' is included in both the preserve_none check and the False return conditions. This might lead to unexpected behavior. Consider removing 'none' from the False condition.

Code Suggestion
```suggestion
        if value is "":
            return default
        if value.lower() == 'true':
            return True
        elif value.lower() == 'false':
            return False
```